### ISSUE TITLE
Menu: Nested submenu (details/summary) has strange behaviour using enter key

### ISSUE THREAD

**Issue by juleskuehn** [wet-boew/wet-boew#8418]
To replicate:
http://wet-boew.github.io/wet-boew/demos/menu/menu-en.html

Open the "Section 3" submenu. The tree submenu, implemented with details/summary behaves as expected using the arrow keys. However, try pressing Enter to expand a submenu. It doesn't work, but does have some visual effect. Press the down key to navigate to focus another submenu. Press Enter again. The _previously_ selected submenu expands!

One solution for this is to disable the enter key within the submenus. Another is to force its behaviour to be identical to the spacebar - behaving like an accordion menu.

**Comment by juleskuehn** [comment]
The spacebar actually expands the submenu but does not focus the first element - meaning you have to use the arrows to enter that submenu even after pressing space. So I think there are two options to make the behaviour consistent:
1. Enter and space behave the same as right arrow.
or
2. Disable enter and space.

**Comment by duboisp** [comment]
What the ARIA Practice 1.1 keyboard interaction model said? https://www.w3.org/TR/wai-aria-practices-1.1/#menu

**Comment by duboisp** [comment]
> Enter:
> * When focus is on a menuitem that has a submenu, opens the submenu and places focus on its first item.
> * Otherwise, activates the item and closes the menu.
>
> Space:
> * (Optional): When focus is on a menuitemcheckbox, changes the state without closing the menu.
> * (Optional): When focus is on a menuitemradio that is not checked, without closing the menu, checks the focused menuitemradio and unchecks any other checked menuitemradio element in the same group.
> * (Optional): When focus is on a menuitem that has a submenu, opens the submenu and places focus on its first item.
> * (Optional): When focus is on a menuitem that does not have a submenu, activates the menuitem and closes the menu.

**Comment by juleskuehn** [comment]
Yes, I was referring to this spec, and that is the exact behaviour I have implemented.

But, the enter and spacebar do not work in detail/summary (nested) submenus in Internet Explorer, and I don't think there is anything I can do about that.

**Comment by thekodester** [comment]
I don't think there much that can be done with Internet Explorer without some messy JS hacks. Considering the fact Microsoft has given up on IE I think it time to just let it die.

**Comment by duboisp** [comment]
> The enter and spacebar do not work in detail/summary (nested) submenus in Internet Explorer

This is might because IE11 run the [details/summary polyfill](https://github.com/wet-boew/wet-boew/blob/master/src/polyfills/details/details.js). See how that polyfill is implemented, you might be able to find a solution.

> Considering the fact Microsoft has given up on IE I think it time to just let it die

Unfortunately IE11 is still highly used. See the [web stats](https://wet-boew.github.io/wet-boew-documentation/research/2018-2-browser-usage.html) that I summarized last March. I didn't get the authorization to publish the exact percentage, but I can say that IE11 have a considerable share greater than 5%.

**Comment by thekodester** [comment]
If you look here https://www.w3counter.com/trends. IE on 3/18 and 4/18 was 2.9% and Edge was around 2.5%.

According to there trends IE is continually dropping.

**Comment by duboisp** [comment]
I agree that IE usage is dropping, but that is not the current reality for our central GC website.

**Comment by thekodester** [comment]
Touche...

**Comment by juleskuehn** [comment]
Fixed by PR #8424 . Closing.
